src/ray/core_worker/core_worker.cc

```diff
@@ -2074,7 +2075,9 @@ Status CoreWorker::KillActor(const ActorID &actor_id, bool force_kill, bool no_r
       cb(Status::Invalid(stream.str()));
     }
   });
-  return f.get();
+  const auto &status = f.get();
+  actor_manager_->InvalidateActorHandleCache(actor_id);
```

Comment:
this leaks the implementation detail. should we rename it to a more general function, such as (OnActorKilled) or somethign?

Reply:
Hmm yeah. I thought about InvalidateActorHandle, and noticed it didn't make sense, so I just changed this. OnActorKilled sounds good.